src/grammar/base.pegjs

```diff
@@ -35,7 +35,7 @@ description = "\"" _ inner:(!"\"" i:. {return i})* "\"" {
 }
 
 // version
-semver = semver:([0-9]+ "." [0-9]+ "." [0-9]+) { return semver.flat().join(""); }
+semver = semver:([0-9]+ "." [0-9]+ "." [0-9]+ ("-" [a-zA-Z0-9.-]+)?) { return semver.flat().flat().join(""); }
```

Comment:
what a ugly flat here.
The flat() method of Array can specific a depth as you want.
flat(2), flat(Infinity)